caddyhttp/proxy/reverseproxy.go

```diff
@@ -318,30 +318,61 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, outreq *http.Request,
 		}
 		pooledIoCopy(backendConn, conn)
 	} else {
+		// NOTE:
+		//   Closing the Body involves acquiring a mutex, which is a
```

Comment:
The Go stdlib uses a Mutex for locking the `res.Body` when calling `Close()` on it. See [here](https://github.com/golang/go/blob/66b5a2f3f0b1d94f69763114a85a977f5bc0277a/src/net/http/transfer.go#L905-L906). I thought that having a bit additional logic in our code is a okay-ish cost considering that the above `defer` will never ever once do anything meaningful until that very rare occurence that our code panics.